models.py

```diff
@@ -363,23 +352,22 @@ def _compute_emissions(self,
 
         # apply Bidirectional LSTM
         if self.uses_lstm:
-            seq_lengths = torch.sum(mask.int(),dim = 1)
+            if mask is None:
+                # shape (batch_size, seq_length)
+                mask = self._get_mask_for(words)   
+            
+            seq_lengths = torch.sum(mask.int(), dim=1)
```

Comment:
Could you please add `# shape:` comments before any lines whose LHS is a new tensor? (e.g. this line, line 366, etc.) It's rather hard to keep track of all the shapes without such comments.